apps/sms/views/inbox/js/inbox.js

```diff
@@ -806,29 +708,16 @@ var InboxView = {
     Threads.delete(threadId);
 
     // Cleanup the DOM
-    this.removeConversationDOM(threadId);
+    var index = this.listData.findIndex((data) => data.id === threadId);
+    if (index > -1) {
+      this.listData.splice(index, 1);
+      this.list.setModel(this.listData);
```

Comment:
yeah you're right, we'll need to re-localize data source and setModel again. I think Wilson already explained the reason why we don't use l10n-id in fast-list. So that's the overhead that we need to deal with once we confirm to land fast-list.
Sorry I'm not very familiar with RxJS, but the methods provide by datasource seems more powerful than native array, and the key point is you can centralize the data changes and using throttle to handle all the setModel/rendering. It would be more clean and readable. Maybe I can try to find some other native approach to reach this idea.